app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/NotificationTest.kt

```diff
@@ -21,91 +18,104 @@ import com.github.se.icebreakrr.ui.navigation.Route
 import com.github.se.icebreakrr.ui.navigation.TopLevelDestinations
 import com.google.firebase.Timestamp
 import com.google.firebase.auth.FirebaseAuth
-import com.google.firebase.firestore.GeoPoint
+import com.google.firebase.functions.FirebaseFunctions
 import java.util.Calendar
+import kotlinx.coroutines.flow.MutableStateFlow
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
-import org.mockito.Mockito
+import org.mockito.Mockito.mock
 import org.mockito.Mockito.`when`
 import org.mockito.kotlin.any
 import org.mockito.kotlin.verify
 
 class NotificationTest {
   private lateinit var navigationActions: NavigationActions
-  private lateinit var mockProfilesRepository: ProfilesRepository
   private lateinit var mockPPRepository: ProfilePicRepository
   private lateinit var profilesViewModel: ProfilesViewModel
+  private lateinit var mockProfilesRepository: ProfilesRepository
+  private lateinit var functions: FirebaseFunctions
+  private lateinit var mockAuth: FirebaseAuth
+  private lateinit var meetingRequestViewModel: MeetingRequestViewModel
 
   @get:Rule val composeTestRule = createComposeRule()
 
   @Before
   fun setUp() {
-    navigationActions = Mockito.mock(NavigationActions::class.java)
-    mockProfilesRepository = Mockito.mock(ProfilesRepository::class.java)
-    mockPPRepository = Mockito.mock(ProfilePicRepository::class.java)
-    profilesViewModel =
-        ProfilesViewModel(mockProfilesRepository, mockPPRepository, FirebaseAuth.getInstance())
+    navigationActions = mock(NavigationActions::class.java)
+    mockPPRepository = mock(ProfilePicRepository::class.java)
+    mockProfilesRepository = mock(ProfilesRepository::class.java)
+    mockAuth = mock(FirebaseAuth::class.java)
+    profilesViewModel = ProfilesViewModel(mockProfilesRepository, mockPPRepository, mockAuth)
+    functions = mock(FirebaseFunctions::class.java)
+    meetingRequestViewModel = MeetingRequestViewModel(profilesViewModel, functions)
+
+    // Mock repository state flows
+    `when`(mockProfilesRepository.isWaiting).thenReturn(MutableStateFlow(false))
+    `when`(mockProfilesRepository.waitingDone).thenReturn(MutableStateFlow(false))
+
+    // Mock successful connection check
+    `when`(mockProfilesRepository.getProfilesInRadius(any(), any(), any(), any())).thenAnswer {
+        invocation ->
+      val onSuccessCallback = invocation.getArgument<(List<Profile>) -> Unit>(2)
+      onSuccessCallback(emptyList())
+      null
+    }
+
+    val profile = mockProfile()
+    `when`(mockProfilesRepository.getMultipleProfiles(any(), any(), any())).thenAnswer { invocation
+      ->
+      val onSuccessCallback = invocation.getArgument<(List<Profile>) -> Unit>(1)
+      onSuccessCallback(listOf(profile))
+    }
 
     `when`(navigationActions.currentRoute()).thenReturn(Route.NOTIFICATIONS)
```

Comment:
Since those tests are incomplete, i thought about using those values like mockPPRepository, mockAuth and function later on, because they ll be needed but i can do as you ask as a temporary solution that is more clean